### DISCUSSION TITLE
Conflicting symbols for timer

### DISCUSSION THREAD

**Discussion by BeardPower** [unikraft/unikraft#267] [Q&A]
Hi there,

I was trying to use lib-musl for some tests.
Unfortunately it clashes with uktimer (it includes timer headers from musl).
/usr/bin/ld: /home/beardpower/.unikraft/apps/app-helloworld/build/libmusl.o: in function `timer_create':
/home/beardpower/.unikraft/apps/app-helloworld/build/libmusl/origin/musl-1.1.19//src/time/timer_create.c:82: multiple definition of `timer_create'; /home/beardpower/.unikraft/apps/app-helloworld/build/libuktime.o:/home/beardpower/.unikraft/unikraft/lib/uktime/timer.c:42: first defined here

scandir has the same issue:
/home/beardpower/.unikraft/apps/app-helloworld/build/libvfscore.o:/home/beardpower/.unikraft/unikraft/lib/vfscore/main.c

Is this a known issue? I inline the code, but the errors did not disappear. Would it be a better approach to define the functions in a header file?

Thanks!

**Comment by nderjung** [comment]
Hi!

cc'ing @razvand @gaulthiergain @cffs and @skuenzer -- these guys are currently working on bringing [musl](https://github.com/unikraft/lib-musl) into stability for Unikraft and can help answer your question. :)

**Comment by razvand** [comment]
Hi, @BeardPower 

Could you share a repository with your build configuration and source code? I'm also cc'ing @dragosargint who worked on Musl and can assist you.

**Reply by BeardPower** [reply]
Certainly!
You can find it here: https://github.com/BeardPower/app-helloworld

**Comment by razvand** [comment]
Everyone, since we are already (viciously) using [the Unikraft Discord server](https://bit.ly/UnikraftDiscord) for discussions, meetings, brainstorming and jokes (mostly bad, we do what we can), we will close the Discussions tab. We will do this on Friday, March 31, 2023.

Please move any discussions or questions on [the Unikraft Discord server](https://bit.ly/UnikraftDiscord). See you all there! :handshake:
